src/lib/viewers/media/Settings.js

```diff
@@ -445,11 +450,14 @@ class Settings extends EventEmitter {
      *
      * @param {string} type - of menu option
      * @param {string} value - of menu option
+     * @param {boolean} setCache - should we set the cache with the given settings value
      * @return {void}
      */
-    chooseOption(type, value) {
-        // Save the value
-        this.cache.set(`media-${type}`, value, true);
+    chooseOption(type, value, setCache = true) {
+        // Save the value if specified
+        if (setCache) {
```

Comment:
Added this so we don't set the cache if we are choosing sd because we have no other options. Now if switch to an HD video we can retain our previous quality setting value